src/irr/asset/COBJMeshFileLoader.cpp

```diff
@@ -122,145 +493,157 @@ asset::SAssetBundle COBJMeshFileLoader::loadAsset(io::IReadFile* _file, const as
+                            submeshes.back()->setPipeline(core::smart_refctd_ptr_static_cast<ICPURenderpassIndependentPipeline>(pipeln.second->clone(0u)));
+                            auto metadata = static_cast<const CMTLPipelineMetadata*>(pipeln.second->getMetadata());
+                            memcpy(
+                                submeshes.back()->getPushConstantsDataPtr()+pipeln.second->getLayout()->getPushConstantRanges().begin()[0].offset,
+                                &metadata->getMaterial().std140PackedData,
+                                sizeof(CMTLPipelineMetadata::SMtl::std140PackedData)
+                            );
+                            //also make/find descriptor set
+                            std::string dsCacheKey = pipeln.first + "?" + mtlName + "?_ds";
+                            types[0] = IAsset::ET_DESCRIPTOR_SET;
+                            auto ds_bundle = _override->findCachedAsset(dsCacheKey, types, ctx.inner, _hierarchyLevel+ICPUMesh::DESC_SET_HIERARCHYLEVELS_BELOW);
+                            auto ds_bundle_contents = ds_bundle.getContents();
+                            core::smart_refctd_ptr<ICPUDescriptorSet> ds3;
+                            if (ds_bundle_contents.first != ds_bundle_contents.second)
+                            {
+                                ds3 = core::smart_refctd_ptr_static_cast<ICPUDescriptorSet>(ds_bundle_contents.first[0]);
+                            }
+                            else
+                            {
+                                auto relDir = (io::IFileSystem::getFileDir(pipeln.first.c_str()) + "/");
+                                images_set_t images = loadImages(relDir.c_str(), metadata->getMaterial(), _hierarchyLevel + ICPUMesh::IMAGE_HIERARCHYLEVELS_BELOW);
+                                ds3 = makeDescSet(images, pipeln.second->getLayout()->getDescriptorSetLayout(3u));
+
+                                if (ds3)
+                                {
+                                    SAssetBundle bundle{ds3};
+                                    _override->insertAssetIntoCache(bundle, dsCacheKey, ctx.inner, _hierarchyLevel+ICPUMesh::DESC_SET_HIERARCHYLEVELS_BELOW);
+                                }
+                            }
+                            submeshes.back()->setAttachedDescriptorSet(std::move(ds3));
+                        }
+
+                        SAssetBundle bundle{submeshes.back()};
+                        _override->insertAssetIntoCache(bundle, genKeyForMeshBuf(ctx, _file->getFileName().c_str(), mtlName, grpName), ctx.inner, _hierarchyLevel+ICPUMesh::MESHBUFFER_HIERARCHYLEVELS_BELOW);
+                        interm_setAssetMutable(AssetManager, submeshes.back().get(), true); //insertion into cache makes inserted asset immutable, so temporarily make it mutable because it has to be adjusted more
```

Comment:
insert at the very end, multithreaded work might fuck up completely with this mutable crap, assets shouldnt be put in cache until they're in their final form.